api/organisations/invites/views.py

```diff
@@ -85,6 +94,16 @@ class InviteLinkViewSet(
     def get_queryset(self):
         organisation_pk = self.kwargs.get("organisation_pk")
         user = self.request.user
+        organisation = Organisation.objects.get(id=organisation_pk)
+        subscription_metadata = organisation.subscription.get_subscription_metadata()
+
+        if (
+            len(settings.AUTO_SEAT_UPGRADE_PLANS) > 0
+            and organisation.num_seats >= subscription_metadata.seats
+            and not organisation.subscription.can_auto_upgrade_seats
+        ):
+            raise SubscriptionDoesNotSupportSeatUpgrade()
```

Comment:
DRY! Can you extract this somewhere else?

Reply:
Done. Sorry, it seems like I left behind work in progress.